ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/rest/RestTestExecutor.java

```diff
@@ -608,41 +613,32 @@ private void waitForWarmStateStore(
     LOG.info("Timed out waiting for correct response");
   }
 
-private void waitForRunningPushQueries(
-      final List<String> queries
+  private void waitForRunningPersistentQueries(
   ) {
-    for (int i = 0; i != queries.size(); ++i) {
-      final String queryStatement = queries.get(i);
-      waitForRunningPush(queryStatement);
-    }
-  }
-
-  private void waitForRunningPush(
-      final String querySql
-  ) {
-    // Make sure push queries are ready to run if they're counting on running before data is
-    // produced.  This is most relevant for scalable push queries since their underlying
-    // persistent queries must be ready. If they're not, we'll get an error.
-
-    if (!(querySql.contains("EMIT CHANGES"))) {
-      // Not a scalable push query, so not needed
-      return;
-    }
-
     final long threshold = System.currentTimeMillis() + MAX_STATIC_WARM_UP.toMillis();
     while (System.currentTimeMillis() < threshold) {
-      final RestResponse<StreamPublisher<StreamedRow>> resp
-          = restClient.makeQueryRequestStreamed(querySql, null);
+      final RestResponse<KsqlEntityList> resp = restClient.makeKsqlRequest("SHOW QUERIES;");
       if (resp.isErroneous()) {
         final KsqlErrorMessage errorMessage = resp.getErrorMessage();
-        LOG.info("Server responded with an error code to a scalable push query. "
-            + "This could be because the persistent query hasn't started yet"
+        LOG.info("Server responded with an error code to show queries."
             + System.lineSeparator() + errorMessage);
         threadYield();
       } else {
-        resp.getResponse().close();
```

Comment:
why don't we still close the response?

Reply:
Nvm, after chatting with Almog clarified that this is because with the push query we keep getting results until we close, whereas with the `show queries` it is one and done